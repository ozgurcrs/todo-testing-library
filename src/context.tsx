import { createContext } from "react";
import { useTodo } from "./hooks/useTodo";

export const TodoStoreContext = createContext({} as ReturnType<typeof useTodo>);
