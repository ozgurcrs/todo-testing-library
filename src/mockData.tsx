export const mockData = [
  {
    isActive: false,
    text: "text 1",
    avatar:
      "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/475.jpg",
    id: "1",
  },
  {
    isActive: false,
    text: "text 2",
    avatar:
      "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/760.jpg",
    id: "2",
  },
  {
    isActive: false,
    text: "text 3",
    avatar:
      "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/112.jpg",
    id: "3",
  },
  {
    isActive: false,
    text: "text 4",
    avatar:
      "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/73.jpg",
    id: "4",
  },
];
