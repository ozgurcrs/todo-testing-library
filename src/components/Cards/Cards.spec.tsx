import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import React from "react";
import { act } from "react-dom/test-utils";
import { Cards } from ".";
import { TodoStoreContext } from "../../context";
import { mockData } from "../../mockData";

const props = {
  cardList: mockData,
  removeItem: jest.fn(),
  changeStatusOfItem: jest.fn(),
};

const mockContext = {
  cardList: mockData,
  removeItem: jest.fn(),
  addItem: jest.fn(),
  changeStatusOfItem: jest.fn(),
};

const renderContainer = () =>
  render(
    <TodoStoreContext.Provider value={mockContext}>
      <Cards {...props} />
    </TodoStoreContext.Provider>
  );

describe("<Cards />", () => {
  test("component should be render", () => {
    const { container } = renderContainer();

    expect(container).toMatchSnapshot();
  });

  test("card should be defined on the screen", () => {
    /*   renderContainer();

    const getCard = screen.getByTestId("card-1-data-test-id");

    expect(getCard).toBeDefined(); */
  });

  test("button should be defined on the screen", () => {
    /*   renderContainer();

    const getButton = screen.getByTestId("delete-1-data-test-id");

    expect(getButton).toBeDefined(); */
  });

  test("typography should be defined on the screen", () => {
    /*  renderContainer();

    const getTypography = screen.getByTestId("typography-1-data-test-id");

    expect(getTypography).toBeDefined();

    const getText = screen.getByText("text 1");

    expect(getText).toBeDefined(); */
  });

  test("when checkbox click, text should be line-through", async () => {
    renderContainer();
    const getTypography = screen.getByTestId("typography-1-data-test-id");
    expect(getTypography).toBeDefined();

    expect(getTypography).not.toHaveStyle("text-decoration: line-through;");

    const getButton = screen.getByTestId("checkbox-1-data-test-id");

    expect(getButton).toBeDefined();

    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(() => {
      fireEvent.click(getButton);
      expect(getTypography).toHaveStyle("text-decoration: none;");
    });
  });

  test("when delete button click, text shouldn't be defined on screen", async () => {
    const { getByTestId } = renderContainer();
    const getTypography = screen.getByTestId("typography-1-data-test-id");
    expect(getTypography).toBeDefined();

    const deleteButton = screen.getByTestId("delete-1-data-test-id");

    expect(deleteButton).toBeDefined();

    // eslint-disable-next-line testing-library/no-unnecessary-act
    await act(async () => {
      fireEvent.click(deleteButton);
      const text = await screen.findByTestId("typography-1-data-test-id");
      expect(text).not.toBeDefined();
    });
  });
});
