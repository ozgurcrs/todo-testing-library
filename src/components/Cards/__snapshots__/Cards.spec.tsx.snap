// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards /> component should be render 1`] = `
<div>
  <div
    class="MuiBox-root css-1e5o465"
    data-testid="card-1-data-test-id"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      data-testid="checkbox-1-data-test-id"
    >
      <input
        class="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </span>
    <img
      alt=""
      loading="lazy"
      src="https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/475.jpg"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 css-1eqmu9j-MuiTypography-root"
      data-testid="typography-1-data-test-id"
    >
      text 1
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      data-testid="delete-1-data-test-id"
      tabindex="0"
      type="button"
    >
      Delete
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiBox-root css-1e5o465"
    data-testid="card-2-data-test-id"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      data-testid="checkbox-2-data-test-id"
    >
      <input
        class="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </span>
    <img
      alt=""
      loading="lazy"
      src="https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/760.jpg"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 css-1eqmu9j-MuiTypography-root"
      data-testid="typography-2-data-test-id"
    >
      text 2
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      data-testid="delete-2-data-test-id"
      tabindex="0"
      type="button"
    >
      Delete
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiBox-root css-1e5o465"
    data-testid="card-3-data-test-id"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      data-testid="checkbox-3-data-test-id"
    >
      <input
        class="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </span>
    <img
      alt=""
      loading="lazy"
      src="https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/112.jpg"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 css-1eqmu9j-MuiTypography-root"
      data-testid="typography-3-data-test-id"
    >
      text 3
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      data-testid="delete-3-data-test-id"
      tabindex="0"
      type="button"
    >
      Delete
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiBox-root css-1e5o465"
    data-testid="card-4-data-test-id"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      data-testid="checkbox-4-data-test-id"
    >
      <input
        class="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </span>
    <img
      alt=""
      loading="lazy"
      src="https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/73.jpg"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 css-1eqmu9j-MuiTypography-root"
      data-testid="typography-4-data-test-id"
    >
      text 4
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      data-testid="delete-4-data-test-id"
      tabindex="0"
      type="button"
    >
      Delete
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
